shark-graph/src/test/java/shark/LongScatterSetTest.kt

```diff
@@ -1,142 +1,137 @@
 package shark
 
-import org.assertj.core.api.Assertions
 import org.junit.Test
 import shark.LongScatterSetAssertion.Companion.assertThat
-import shark.internal.hppc.HHPC
 import shark.internal.hppc.LongScatterSet
 
 class LongScatterSetTest {
 
-  @Test fun `verify LongScatterSet#add operation works correctly`() {
-    val set = LongScatterSet()
-    runAssertionsForAddOperation(set)
+  @Test fun `new set is empty`() {
+    assertThat(LongScatterSet()).isEmpty()
   }
 
-  @Test fun `verify LongScatterSet#remove operation works correctly`() {
+  @Test fun `adding elements`() {
     val set = LongScatterSet()
-    runAssertionsForRemoveOperation(set)
+
+    TEST_VALUES_LIST.forEach { set.add(it) }
+
+    assertThat(set)
+        .contains(*TEST_VALUES_LIST.toLongArray())
+        .hasSize(TEST_VALUES_LIST.size)
   }
 
-  @Test fun `verify LongScatterSet#release operation works correctly`() {
+  @Test fun `+= operator works as addition`() {
     val set = LongScatterSet()
-    set.add(42)
-    set.release()
+
+    set.add(TEST_VALUE)
 
     assertThat(set)
-        .isEmpty()
-        .doesNotContain(42)
+        .contains(TEST_VALUE)
+        .hasSize(1)
   }
 
-  @Test fun `verify LongScatterSet#ensureCapacity does not break Set when called before those operations`() {
-    val setForAddOperation = LongScatterSet().apply { ensureCapacity(10) }
-    val setForRemoveOperation = LongScatterSet().apply { ensureCapacity(10) }
+  @Test fun `adding element twice doesn't cause duplicates`() {
+    val set = LongScatterSet()
+
+    set.add(TEST_VALUE)
+    set.add(TEST_VALUE)
 
-    // Verify that calling ensureCapacity doesn't affect add/remove operations
-    runAssertionsForAddOperation(setForAddOperation)
-    runAssertionsForRemoveOperation(setForRemoveOperation)
+    assertThat(set)
+        .contains(TEST_VALUE)
+        .hasSize(1)
   }
 
-  @Test fun `verify LongScatterSet#ensureCapacity does not break Set when called after those operations`() {
-    val set = LongScatterSet().apply {
-      add(42)
-      add(10)
-      ensureCapacity(100)
-    }
+  @Test fun `adding elements with matching hash`() {
```

Comment:
maybe "elements with equal hash can be added" ? the test name here says we're adding but doesn't outline the expected result.